src/mlpack/tests/local_coordinate_coding_test.cpp

```diff
@@ -43,15 +46,18 @@ void VerifyCorrectness(const vec& beta, const vec& errCorr, double lambda)
   }
 }
 
-
-TEST_CASE("LocalCoordinateCodingTestCodingStep",
-          "[LocalCoordinateCodingTest]")
+TEMPLATE_TEST_CASE("LocalCoordinateCodingTestCodingStep",
+    "[LocalCoordinateCodingTest]", arma::mat, arma::fmat)
 {
+  typedef TestType MatType;
+  typedef arma::Col<typename MatType::elem_type> VecType;
+
   double lambda1 = 0.1;
   uword nAtoms = 10;
 
-  mat X;
-  X.load("mnist_first250_training_4s_and_9s.arm");
+  mat inX; // The .arm file is saved as an arma::mat.
+  inX.load("mnist_first250_training_4s_and_9s.arm");
```

Comment:
Never heard of this extension before, :smile:
@conradsnicta should patent it

Reply:
Data saved in Armadillo's native arma_binary format is platform dependent.  It's a "quick-n-dirty" binary format that's meant to be more efficient than CSV or ascii, and less finicky than HDF5. This comes at the cost of being platform and/or OS dependent.  For example, stuff saved on an ARM machine may not load on an x86-64 machine and vice-versa.

This is not a problem in many cases, but it's probably not a good idea to have it as part of official mlpack tests, which are meant to work on many platforms.  For better portability, suggest to use the CSV format instead.

Reply:
That's a good point.  In 2c035b56, I converted the test data to be stored in .csv instead of the Armadillo binary format.  It's very slightly larger in the `.tar.bz2` (like 20kb), but I don't think that's a big deal.